packages/paste-website/src/components/shortcodes/package-status-legend/index.tsx

```diff
@@ -1,70 +1,83 @@
 import * as React from 'react';
 import {Box} from '@twilio-paste/box';
-import {Text} from '@twilio-paste/text';
-import {Stack} from '@twilio-paste/stack';
-import {SuccessIcon} from '@twilio-paste/icons/esm/SuccessIcon';
-import {AssetStatus} from '../../component-status/AssetStatus';
-import {PeerReviewStatus} from '../../component-status/PeerReviewStatus';
-import type {PackageStatusObject} from '../../../utils/types';
+import {Badge} from '@twilio-paste/badge';
+import type {BadgeVariants} from '@twilio-paste/badge/src/types';
+import {useTheme} from '@twilio-paste/theme';
+
+import {NewIcon} from '@twilio-paste/icons/esm/NewIcon';
+import {ProcessDraftIcon} from '@twilio-paste/icons/esm/ProcessDraftIcon';
+
+import {BadgePopoverButton, BadgePopoverContainer, BadgePopover} from '../../badgePopover';
+import {StatusDescriptions} from '../../../constants';
 
 interface PackageStatusLegendProps {
-  packageStatus: PackageStatusObject;
+  packageStatus?: string;
+  figmaStatus?: string;
+  designCommitteeReview?: string;
+  engineerCommitteeReview?: string;
 }
 
-const PackageStatusLegend: React.FC<PackageStatusLegendProps> = ({packageStatus}) => {
-  const {
-    Figma,
-    Documentation,
-    // eslint-disable-next-line @typescript-eslint/naming-convention
-    Design_committee_review,
-    // eslint-disable-next-line @typescript-eslint/naming-convention
-    Engineer_committee_review,
-    Code,
-  } = packageStatus[0].node.data;
+const PackageStatusBadge: React.FC<{status: string}> = ({status}) => {
+  let badgeVariant: BadgeVariants | undefined;
+  let popoverContent = '';
+
+  if (status === 'Alpha') {
+    badgeVariant = 'new';
+    popoverContent = StatusDescriptions.ALPHA;
+  }
+
+  if (status === 'Beta') {
+    badgeVariant = 'info';
+    popoverContent = StatusDescriptions.BETA;
+  }
 
   return (
```

Comment:
Unless the outer fragments are required, could we remove them? I find that functional components are clearest when written most like functions.

```suggestion
if (badgeVariant != null) {
    return (
        <BadgePopoverContainer baseId="status">
            <BadgePopoverButton variant={badgeVariant}>
              <>
                <NewIcon decorative size="sizeIcon10" />
                {status}
              </>
            </BadgePopoverButton>
            <BadgePopover aria-label={status}>{popoverContent}</BadgePopover>
        </BadgePopoverContainer>
    )
}

return null;
```

Also, is the inner fragment required because the button wrapper expects a single child? If so, I think it's worth a comment explaining so folks don't remove it in the future thinking it's superfluous. 

If it's not required👇 

```suggestion
if (badgeVariant != null) {
    return (
        <BadgePopoverContainer baseId="status">
            <BadgePopoverButton variant={badgeVariant}>
                <NewIcon decorative size="sizeIcon10" />
                {status}
            </BadgePopoverButton>
            <BadgePopover aria-label={status}>{popoverContent}</BadgePopover>
        </BadgePopoverContainer>
    )
}

return null;
```

Reply:
Updated.